crates/conversions/src/byte_array.rs

```diff
@@ -68,3 +70,32 @@ impl From<ByteArray> for String {
         format!("{full_words_string}{pending_word_string}")
     }
 }
+
+
+impl fmt::Display for ByteArray {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{self:?}") 
+    }
+}
+
+// we can implement that or dont use Serialize/Deserialize in CallResult/CallFailure in rpc.rs
```

Comment:
Let's delete the derivation of `Serialize`/`Deserialize` in `CallResult`/`CallFailure`, as it is unused anyway.